spec/ics-027-interchain-accounts/README.md

```diff
@@ -50,88 +50,111 @@ It does not restrict how you can distinguish signers that was not made by the co
+
+`tryRunTx` method in `IBCAccountModule` interface defines the way to create an outgoing packet for a specific `type`. `Type` indicates how the IBC account transaction should be constructed and serialized for the host chain. Generally, `type` indicates what blockchain framework the host chain was built on.
+
+`createAccount` defines the way to determine the account's address by using the packet. If the host chain doesn't support a deterministic way to generate an address with data, it can be generated using the internal logic of the host chain. A newly created interchain account must not conflict with an existing account. Therefore, the host chain (on that the interchain account lives in) must keep track of which blockchains have created an interchain account within the host chain in order to verify the transaction signing authority in `authenticateTx`.
+
+`authenticateTx` validates the transaction and checks that the signers in the transaction have the right permissions. `
+
+`runTx` executes a transaction after the transaction has been successfully authenticated.
 
 ```typescript
 type Tx = object
 
 interface IBCAccountModule {
-  createOutgoingPacket(chainType: Uint8Array, data: any)
-  createAccount(address: Uint8Array)
-  generateAddress(identifier: Identifier, salt: Uint8Array): Uint8Array
+  tryRegisterIBCAccount(data: Uint8Array)
+  tryRunTx(chainType: Uint8Array, data: any)
+  createAccount(packet: Packet, data: Uint8Array): Address
   deserialiseTx(txBytes: Uint8Array): Tx
   authenticateTx(tx: Tx): boolean
-  runTx(tx: Tx): uint32
+  runTx(tx: Tx): Result
 }
 ```
 
-`RegisterIBCAccountPacketData` is used by the counterparty chain to register an account. An interchain account's address is defined deterministically with the channel identifier and salt. The ```generateAccount``` method is used to generate a new interchain account's address. It is recommended to generate address by ```hash(identifier+salt)```, but other methods may be used. This function must generate a unique and deterministic address by utilising identifier and salt.
+`IBCAccountPacketData` specifies the type of packet. The `IBCAccountPacketData.data` is arbitrary and can be used for different purposes depending on the value of `IBCAccountPacketData.type`.
+
+When the value of `type` is `REGISTER`, `IBCAccountPacketData.data` can be used as information by which the account is created on the host chain. When the value of `type` is `RUNTX`, `IBCAccountPacketData.data` contains the tx bytes.
 
 ```typescript
-interface RegisterIBCAccountPacketData {
-  salt: Uint8Array
+// `Type` enumerator defines the packet type.
+enum Type {
```

Comment:
You can probably use `oneof`, see below comment

Reply:
I think that the all fields in this definition can be used mutually, so I think the enumerator is enough.